File: src/android/app/src/main/java/evicentemedina/esperpento/objects/Thread.java
package evicentemedina.esperpento.objects;

import android.support.annotation.NonNull;

import java.util.Date;

public class Thread {
    private int id, votes;
    private String title, content, community, user;
    private Date time, edited;

    // New thread constructor
    public Thread(@NonNull String title, @NonNull String content,
                  @NonNull String community, @NonNull String user) {
        this.title = title;
        this.content = content;
        this.community = community;
        this.user = user;
    }

    // Full constructor
    public Thread(int id, @NonNull String title, @NonNull String content, @NonNull Date time,
                  int votes, Date edited, @NonNull String community, @NonNull String user) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.time = time;
        this.votes = votes;
        this.edited = edited;
        this.community = community;
        this.user = user;
    }

    public int getId() {
        return id;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getTitle() {
        return title;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getContent() {
        return content;
    }

    public int getVotes() {
        return votes;
    }

    public Date getTime() {
        return time;
    }

    public void setEdited(Date edited) {
        this.edited = edited;
    }

    public Date getEdited() {
        return edited;
    }

    public String getCommunity() {
        return community;
    }

    public String getUser() {
        return user;
    }
}

File: src/server/thread.py
class Thread:
    def __init__(self, id, title, content, time, votes, edited, community, user):
        self.id = id
        self.title = title
        self.content = content
        self.time = time
        self.votes = votes
        self.edited = edited
        self.community = community
        self.user = user

    def __init__(self, row):
        self.id = row[0]
        self.title = row[1]
        self.content = row[2]
        self.time = row[3]
        self.votes = row[4]
        self.edited = row[5]
        self.community = row[6]
        self.user = row[7]

    def asList(self):
        return [self.id, self.title, self.content, self.time,
                self.votes, self.edited, self.community, self.user]
